sentinel/src/lib/feed.js

```diff
@@ -1,16 +1,32 @@
+const async = require('async');
 const logger = require('./logger');
 const db = require('../db');
 const metadata = require('./metadata');
 const tombstoneUtils = require('@medic/tombstone-utils');
+const transitionsLib = require('../config').getTransitionsLib();
 
 // we don't run transitions on ddocs or info docs
 const IDS_TO_IGNORE = /^_design\/|-info$/;
 
 const RETRY_TIMEOUT = 60000; // 1 minute
+const PROGRESS_REPORT_INTERVAL = 500; // items
+const MAX_QUEUE_SIZE = 100;
 
-let listener;
 let init;
 let request;
+let processed = 0;
+let lastSeq;
+let lastInQueue = false;
+
+const listener = change => changeQueue.push(change);
```

Comment:
This name isn't right: it's not a listener, it's not listening, right? It's enqueuing a piece of work.